### PR TITLE
Add basic outline source

### PR DESCRIPTION
**Pull request by tamy0612**
Since I'm not sure how to use highlighting stuff, it's just a basic implementation.

### CODE REVIEW

File: rplugin/python3/denite/source/outline.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+class Source(Base):
+
+    def __init__(self, vim):
+        super().__init__(vim)
+
+        self.name = 'outline'
+        self.kind = 'file'
+        self.vars = {
+            'encode': ['utf-8'],
```
Comment (Shougo) [inline rplugin/python3/denite/source/outline.py]: Why the encode is List?

File: rplugin/python3/denite/source/outline.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            'sort': ['no'],
```
Comment (Shougo) [inline rplugin/python3/denite/source/outline.py]: Why this var is needed?
It seems not used.

File: rplugin/python3/denite/source/outline.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            'ignore_types': []
+        }
+
+    def on_init(self, context):
+        context['__bufname'] = context['args'][0] if len(
+            context['args']) > 0 else self.vim.current.buffer.name
+
+    def gather_candidates(self, context):
+        command = ['ctags', '-x', context['__bufname'], '--sort=no']
```
Comment (Shougo) [inline rplugin/python3/denite/source/outline.py]: `--sort=no` is needed?

File: rplugin/python3/denite/source/outline.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
+# ============================================================================
+# FILE: buffer.py
```
Comment (tamy0612) [inline rplugin/python3/denite/source/outline.py]: Fixed

File: rplugin/python3/denite/source/outline.py (hunk 1 above)
Comment (tamy0612) [inline rplugin/python3/denite/source/outline.py]: fixed

File: rplugin/python3/denite/source/outline.py (hunk 2 above)
Comment (tamy0612) [inline rplugin/python3/denite/source/outline.py]: removed

File: rplugin/python3/denite/source/outline.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        self.name = 'outline'
+        self.kind = 'file'
+        self.vars = {
+            'encode': ['utf-8'],
+            'sort': ['no'],
+            'ignore_types': []
+        }
+
+    def on_init(self, context):
+        context['__bufname'] = context['args'][0] if len(
```
Comment (tamy0612) [inline rplugin/python3/denite/source/outline.py]: modified

File: rplugin/python3/denite/source/outline.py (hunk 3 above)
Comment (tamy0612) [inline rplugin/python3/denite/source/outline.py]: put `vars['command']` and `vars['options']`

**Comment by tamy0612** [comment]
Also, I added some codes for highlight, while it might be a bit too much though...

**Comment by Shougo** [comment]
Please add the documentation.

**Comment by tamy0612** [comment]
Could you check it?

File: doc/denite.txt [hunk 6]
```diff
@@ -995,7 +995,21 @@ menu		Gather custom menus.
...
+		1. filename (use current buffer path if omitted)
+
+		Source custom variables:
+		command 	the default command to get outline
+				(default: ['ctags'])
+		options 	the default options for command
+				(default: ['-x'])
+		ignore_types 	the list of ignore types
+				(default: [])
+		encode 		the encode
```
Comment (Shougo) [inline doc/denite.txt]: It should be `encoding` instead of `encode`.

File: rplugin/python3/denite/source/outline.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+            'options': ['-x'],
+            'ignore_types': [],
+            'encode': 'utf-8'
+        }
+
+    def on_init(self, context):
+        context['__path'] = context['args'][0] if len(
+            context['args']) > 0 else self.vim.current.buffer.name
+
+    def highlight_syntax(self):
```
Comment (Shougo) [inline rplugin/python3/denite/source/outline.py]: It should be highlight.

File: rplugin/python3/denite/source/outline.py [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+
+class Source(Base):
+
+    def __init__(self, vim):
+        super().__init__(vim)
+
+        self.name = 'outline'
+        self.kind = 'file'
+        self.syntax_name = 'deniteSource_outline'
```
Comment (Shougo) [inline rplugin/python3/denite/source/outline.py]: You don't have to define the syntax name.

**Comment by Shougo** [comment]
LGTM

**Pull request merged**